### ISSUE TITLE
build failure for tensorflow v1.12.0 

### ISSUE THREAD

**Issue by starkdg** [tensorflow/tensorflow#24527]
Hello.  Having a problem with building tensorflow from source.  The most recent version I've been able to compile was 1.9.0, and that only after multiple attempts.  Here is the result from my recent attempt at building the build_pip_package target.  Any help here would be greatly appreciated.  Just let me know if I left out any valuable information, I'll be more than happy to provide it.  Thanks. 

**System information**

Distributor ID:	Debian
Description:	Debian GNU/Linux 9.6 (stretch)
Release:	9.6
Codename:	stretch

- TensorFlow installed from (source or binary): source
- TensorFlow version: v1.12.0 from git repos.
- Python version: 3.5.3
- Installed using virtualenv? pip? conda?: virtualenv or system
- Bazel version (if compiling from source): v0.21.0
- GCC/Compiler version (if compiling from source): gcc 4.8.4
- CUDA/cuDNN version: none 
- GPU model and memory: none. (cpu only)
**Describe the problem**
Unable to compile pip package

**Provide the exact sequence of commands / steps that you executed before running into the problem**

./configure   (enter default settings)

bazel build --config=opt //tensorflow/tools/pip_package:build_pip_package

**Any other info / logs**

INFO: Invocation ID: b0ec9363-a8cc-4c61-9cbe-64b57af62a73
ERROR: /home/david/.cache/bazel/_bazel_david/1f9a24d051ca8bc02465362202ebba4c/external/protobuf_archive/BUILD:591:1: Traceback (most recent call last):
	File "/home/david/.cache/bazel/_bazel_david/1f9a24d051ca8bc02465362202ebba4c/external/protobuf_archive/BUILD", line 591
		internal_gen_well_known_protos_java(srcs = WELL_KNOWN_PROTOS)
	File "/home/david/.cache/bazel/_bazel_david/1f9a24d051ca8bc02465362202ebba4c/external/protobuf_archive/protobuf.bzl", line 269, in internal_gen_well_known_protos_java
		Label(("%s//protobuf_java" % REPOSITOR...))
	File "/home/david/.cache/bazel/_bazel_david/1f9a24d051ca8bc02465362202ebba4c/external/protobuf_archive/protobuf.bzl", line 269, in Label
		REPOSITORY_NAME
The value 'REPOSITORY_NAME' has been removed in favor of 'repository_name()', please use the latter (https://docs.bazel.build/versions/master/skylark/lib/native.html#repository_name). You can temporarily allow the old name by using --incompatible_package_name_is_a_function=false
ERROR: /home/david/.cache/bazel/_bazel_david/1f9a24d051ca8bc02465362202ebba4c/external/protobuf_archive/BUILD:715:1: Target '@protobuf_archive//:use_fast_cpp_protos' contains an error and its package is in error and referenced by '@protobuf_archive//:protobuf_python'
ERROR: /home/david/tensorflow/tensorflow/python/BUILD:3546:1: Target '@protobuf_archive//:protobuf_python' contains an error and its package is in error and referenced by '//tensorflow/python:util'
ERROR: Analysis of target '//tensorflow/tools/pip_package:build_pip_package' failed; build aborted: Analysis failed
INFO: Elapsed time: 0.447s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (2 packages loaded, 15 targets configured)
    currently loading: tensorflow/core

**Comment by starkdg** [comment]
I was able to get past that error by passing "--incompatible_package_name_is_a_function=false" flag (as recommended in previous error messages).  The compilation proceeded quite a bit further, only to get the following.  

ERROR: /home/david/.cache/bazel/_bazel_david/1f9a24d051ca8bc02465362202ebba4c/external/protobuf_archive/BUILD:70:1: C++ compilation of rule '@protobuf_archive//:protobuf_lite' failed (Exit 1) gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer '-std=c++0x' -MD -MF ... (remaining 32 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
In file included from /usr/include/c++/4.8/random:50:0,
                 from /usr/include/c++/4.8/bits/stl_algo.h:65,
                 from /usr/include/c++/4.8/algorithm:62,
                 from external/protobuf_archive/src/google/protobuf/stubs/common.h:38,
                 from external/protobuf_archive/src/google/protobuf/stubs/int128.h:33,
                 from external/protobuf_archive/src/google/protobuf/stubs/int128.cc:31:
/usr/include/c++/4.8/bits/random.h: In constructor 'std::student_t_distribution<_RealType>::student_t_distribution(const std::student_t_distribution<_RealType>::param_type&)':
/usr/include/c++/4.8/bits/random.h:3391:39: internal compiler error: in cp_parser_lookup_name, at cp/parser.c:21173
       : _M_param(__p), _M_nd(), _M_gd(__p.n() / 2, 2)
                                       ^
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-4.8/README.Bugs> for instructions.
The bug is not reproducible, so it is likely a hardware or OS problem.
Target //tensorflow/tools/pip_package:build_pip_package failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 392.793s, Critical Path: 61.91s
INFO: 815 processes: 1 local, 814 processwrapper-sandbox.
FAILED: Build did NOT complete successfully

**Comment by lxl910915** [comment]
We build successfully using bazel 0.18.0 and gcc 5.4.

**Comment by msymp** [comment]
Hello @starkdg, In response to your original question, you can reliably build from r1.12 sources, with **_Python 3.5.3 and gcc 4.8.4 and bazel 0.15.0_**
Bazel is generally behind the curve to the latest TF distros.
--

**Comment by starkdg** [comment]
Thanks @msymp for the response.  I'm pretty sure I tried that already, but I just tried it again just to be sure.  While the compilation does advance much further, it ends with a a different error.  To be sure, the missing dependency information is gone, but  a seg fault.  

python 3.5.3
gcc 4.8.4
bazel 0.15.0
tensorflow v1.12.0 source
system: Linux debian 4.9.0-8-amd64 #1 SMP Debian 4.9.130-2 (2018-10-27) x86_64 GNU/Linux

Error log:
ERROR: /home/david/.cache/bazel/_bazel_david/1f9a24d051ca8bc02465362202ebba4c/external/llvm/BUILD.bazel:1118:1: C++ compilation of rule '@llvm//:code_gen' failed (Exit 1)
In file included from external/llvm/include/llvm/Support/Options.h:41:0,
                 from external/llvm/include/llvm/IR/LLVMContext.h:21,
                 from external/llvm/include/llvm/IR/Metadata.h:30,
                 from external/llvm/include/llvm/IR/TrackingMDRef.h:17,
                 from external/llvm/include/llvm/IR/DebugLoc.h:18,
                 from external/llvm/include/llvm/IR/Instruction.h:22,
                 from external/llvm/include/llvm/IR/BasicBlock.h:23,
                 from external/llvm/include/llvm/IR/Function.h:28,
                 from external/llvm/include/llvm/IR/CallSite.h:34,
                 from external/llvm/include/llvm/Analysis/MemoryLocation.h:21,
                 from external/llvm/include/llvm/Analysis/AliasAnalysis.h:44,
                 from external/llvm/include/llvm/CodeGen/MachineInstr.h:24,
                 from external/llvm/include/llvm/CodeGen/MachineBasicBlock.h:22,
                 from external/llvm/lib/CodeGen/PHIEliminationUtils.h:13,
                 from external/llvm/lib/CodeGen/PHIElimination.cpp:16:
external/llvm/include/llvm/Support/CommandLine.h: In constructor 'llvm::cl::opt<DataType, ExternalStorage, ParserClass>::opt(const Mods& ...) [with Mods = {char [32], llvm::cl::initializer<bool>, llvm::cl::OptionHidden, llvm::cl::desc}; DataType = bool; bool ExternalStorage = false; ParserClass = llvm::cl::parser<bool>]':
external/llvm/include/llvm/Support/CommandLine.h:1363:50: internal compiler error: Segmentation fault
       : Option(Optional, NotHidden), Parser(*this) {
                                                  ^

**Comment by starkdg** [comment]
Alright, I was finally able to build the r1.12.0 package under the following setup combination.  The usual error ("internal compiler error: segfault") still occured, but I found that if I just rerun the command, the build eventually completes successfully.  It took 3 or 4 tries before eventually succeeding.   Not a very satisfying or comforting resolution, but at least it's something.  

Not that it matters, but I tried to compile in a virtualenv created with pipenv.
python 3.5.3
gcc 4.8.4
bazel 0.15.0
tensorflow v1.12.0 source
system: Linux debian 4.9.0-8-amd64 #1 SMP Debian 4.9.130-2 (2018-10-27) x86_64 GNU/Linux
